src/app/core/user/user.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+export class User implements IUser{
```

Comment:
It's ok for now. But keep in mind that we should rethink it in the future.

- the type of implemented interface
- the necessity of this service
- a place for this service

Same for Course class